### PR TITLE
Remove all inter-feature dependencies for the 'events' feature

### PR DESCRIPTION
**Pull request by Tyratox**
This PR removes the inter-feature dependencies from the 'events' feature.

One of the major changes is that the EventReducer no longer stores all information regarding the different event types but just information about what type it is, its id and a start and end time. The different features are then required to handle everything else by themselves such as storing the concrete state for the different events.

Just as an information: This PR was horrible to do because a lot of changes in many places were required. I assume reviewing it will be even worse so yeah.. Sorry for that but I did not manage to split it into multiple PRs that make sense by themselves.

### CODE REVIEW

File: fe1-web/src/features/wallet/objects/Token.ts
Comment (pierluca) [inline fe1-web/src/features/wallet/objects/Token.ts:51]: nice catch !

File: fe1-web/src/features/wallet/objects/Token.ts
Comment (pierluca) [inline fe1-web/src/features/wallet/objects/Token.ts:79]: outside the scope of this PR: typed errors would be nice here, to distinguish between line 73, a "this-should-never-happen-and-all-our-expectations-have-been-violated" error, and line 79, a "tough-luck-you-have-no-token" kind of error

**Review by pierluca — COMMENTED** [review]
Phew, this was a long one ! Pretty good overalls, only minor remarks, though I haven't been able to review everything in-depth.

@Xelowak I'll let you have a more in-depth check of the reducers and their associated tests (and approve if you're okay with them), I haven't reviewed those in detail.

File: fe1-web/src/features/wallet/interface/Feature.ts
Comment (Tyratox) [reply fe1-web/src/features/wallet/interface/Feature.ts:27]: No we don't, you are right :)

File: fe1-web/src/features/wallet/objects/DummyWallet.ts
Comment (Tyratox) [reply fe1-web/src/features/wallet/objects/DummyWallet.ts:7]: 👍

File: fe1-web/src/features/wallet/objects/Token.ts
Comment (Tyratox) [reply fe1-web/src/features/wallet/objects/Token.ts:79]: Yes I agree!

File: fe1-web/src/features/rollCall/hooks/RollCallHooks.ts
Comment (Tyratox) [reply fe1-web/src/features/rollCall/hooks/RollCallHooks.ts]: I rewrote the function using reduce and named functions :) I hope it is easier to read now?

File: fe1-web/src/features/evoting/reducer/ElectionReducer.ts
Comment (Xelowak) [inline fe1-web/src/features/evoting/reducer/ElectionReducer.ts:24]: Is this related to any LAO or is it completely generic and stores all elections?
The same questions applies for each <Event>Reducer.

File: fe1-web/src/features/rollCall/hooks/RollCallHooks.ts
Comment (Xelowak) [reply fe1-web/src/features/rollCall/hooks/RollCallHooks.ts]: It is!

File: fe1-web/src/features/rollCall/reducer/RollCallReducer.ts
Comment (Xelowak) [inline fe1-web/src/features/rollCall/reducer/RollCallReducer.ts:168]:
```suggestion
 * Retrieves a rollcall by its id from the redux store
```

File: fe1-web/src/features/rollCall/reducer/__tests__/RollCallReducer.test.ts
Comment (Xelowak) [inline fe1-web/src/features/rollCall/reducer/__tests__/RollCallReducer.test.ts:50]:
```suggestion
    it('throws an error if the store already contains a rollcall with the same id', () => {
```

**Review by Xelowak — COMMENTED** [review]
Good job!
As pierluca asked me, I checked reducers and their tests and everything seems fine to me. I just had some questions for you.

File: fe1-web/src/features/evoting/reducer/ElectionReducer.ts
Comment (Tyratox) [reply fe1-web/src/features/evoting/reducer/ElectionReducer.ts:24]: Yes this is supposed to be generic and store all elections. The ids of the different elections, roll calls, etc. are obtained by computing a hash function on some constants, the laoId and some other properties. Thus as long as the hash function is collision resistant (which it should be, otherwise we have problems with the security-related parts of the protocol), the ids will be unique. And as you said, if you would like to obtain all elections for a given lao, you can use the events reducer to obtain the set of ids and then load them from the election reducer. As far as I remember, this is only done for roll calls in the wallet and in all other placed we do not need a list of a given event type for a given lao id.

Comment (Xelowak) [reply fe1-web/src/features/evoting/reducer/ElectionReducer.ts:24]: Okay, thanks for the explanation. This implementation seems fine to me.

**Review by Xelowak — APPROVED** [review]

**Pull request merged**
